src/app/(admin)/admin/_components/Specialist/Contacts.js

```diff
@@ -2,20 +2,40 @@ import React from 'react';
 import { FormFieldWrapper } from '@admin/components/FormFieldWrapper';
 import { SpecialistFormFields, SpecialistFormSections } from '@admin/_lib/specialistData';
 import { TextInputList } from '@admin/components/TextInputList';
+import { useWatch } from 'react-hook-form';
+import { DisplayMatchingEntity } from '@admin/components/DisplayMatchingEntity';
+import { useFindMatchingEntities } from '@admin/_hooks/common';
 
 export function Contacts() {
-  const { phone, email, website, instagram, facebook, youtube, linkedin, tiktok, viber, telegram } =
-    SpecialistFormFields;
+  const PHONE = 'phone';
+  const EMAIL = 'email';
+
+  const { phone, email, website } = SpecialistFormFields;
   const contactsList = [phone, email, website];
-  const socialMediaList = [instagram, facebook, youtube, linkedin, tiktok, viber, telegram];
+
+  const currentPhone = useWatch({ name: PHONE });
+  const currentEmail = useWatch({ name: EMAIL });
+
+  const { data: entitiesMatchingPhone, num: numEntitiesMatchingPhone } = useFindMatchingEntities({
+    key: PHONE,
+    value: currentPhone,
+  });
+  const { data: entitiesMatchingEmail, num: numEntitiesMatchingEmail } = useFindMatchingEntities({
+    key: EMAIL,
+    value: currentEmail,
+  });
+
+  const hasEntitiesMatchingPhone = currentPhone && numEntitiesMatchingPhone > 0;
+  const hasEntitiesMatchingEmail = currentEmail && numEntitiesMatchingEmail > 0;
 
   return (
-    <FormFieldWrapper title={SpecialistFormSections.contacts} className="mt-3">
+    <FormFieldWrapper title={SpecialistFormSections.contacts}>
       <div className="flex w-full flex-col md:flex-row md:gap-6 [&>*]:flex-grow">
         <TextInputList textInputList={contactsList} />
       </div>
-      <div className="flex w-full flex-col flex-wrap md:flex-row md:gap-6 [&>*]:flex-grow">
-        <TextInputList textInputList={socialMediaList} className="w-1/3" />
+      <div className="mb-8 flex flex-col gap-6">
+        {hasEntitiesMatchingPhone && <DisplayMatchingEntity entities={entitiesMatchingPhone} label="телефон" />}
+        {hasEntitiesMatchingEmail && <DisplayMatchingEntity entities={entitiesMatchingEmail} label="пошта" />}
```

Comment:
Maybe you can move it to separate component together with hooks and reuse it for organizations as well?

Reply:
Created MatchingEntityList and MatchingEntityListItem to serve the purpose you suggested